app/controllers/messages_controller.rb

```diff
@@ -58,7 +58,8 @@ def create
   def update
     # Clicking edit beneath a message actually submits to create and not here. This action is only used for next/prev conversation.
     # In order to force a morph we PATCH to here and redirect.
-    if @message.update(message_params)
+    @message.allow_deleted_assistant = true # disable that validation
```

Comment:
Now that you're hiding the edit icon on messages within conversations with deleted assistants, can we remove all this special handling for "allow_deleted_assistant"?  (both here and within message.rb)